backend/controllers/prisonerProfile/prisonerLocationHistory.js

```diff
@@ -12,7 +12,37 @@ const {
   app: { notmEndpointUrl: dpsUrl },
 } = require('../../config')
 
-module.exports = ({ prisonApi, logError }) => async (req, res) => {
+const fetchStaffName = (context, staffId, prisonApi) => {
+  return prisonApi
+    .getStaffDetails(context, staffId)
+    .then(staff => formatName(staff.firstName, staff.lastName))
+    .catch('No staff name found')
+}
+
+const fetchReasonDescription = (context, assignmentReasonCode, caseNotesApi) => {
+  return caseNotesApi
+    .getCaseNoteTypes(context)
+    .then(caseNoteTypes => caseNoteTypes.find(type => type.code === 'MOVED_CELL'))
+    .then(cellMoveTypes => {
+      return cellMoveTypes?.subCodes.map(subType => ({
+        value: subType.code,
+        text: subType.description,
+      }))
+    })
+    .then(cellMoveReasonRadioValues => cellMoveReasonRadioValues.find(record => record.value === assignmentReasonCode))
+    .then(assignmentReason => assignmentReason.text)
+    .catch('No reason description found')
+}
+
+const fetchWhatHappened = (context, offenderNo, bookingId, bedAssignmentHistorySequence, prisonApi, whereaboutsApi) => {
+  return whereaboutsApi
+    .getCellMoveReason(context, bookingId, bedAssignmentHistorySequence)
+    .then(cellMoveReason => prisonApi.getCaseNote(context, offenderNo, cellMoveReason.cellMoveReason.caseNoteId))
+    .then(caseNote => caseNote.text)
+    .catch(err => 'Not entered')
```

Comment:
This seems different to how we handle these types of errors elsewhere what was the reason for doing it like this?